framework/db/transaction_manager.py

```diff
@@ -447,20 +447,20 @@ def GetHrtResponse(self, filter_data, trans_id, target_id=None):
         # Data validation
         languages = ['bash']  # Default script language is set to bash.
         if filter_data.get('language'):
-            languages = map(lambda x: x.strip(), filter_data['language'][0].split(','))
+            languages = map(lambda x: x.strip(), filter_data['language'])
 
         proxy = None
         search_string = None
         data = None
 
         if filter_data.get('proxy'):
-            proxy = None if filter_data['proxy'][0] == '' else filter_data['proxy'][0]
+            proxy = filter_data['proxy'][0]
```

Comment:
Looks cleaner that way. Thanks for updating! The best would have been to have `proxy = filter_data.get('proxy', None)` but since Tornado provides a list, it doesn't seem that straightforward.